common/types/double.go

```diff
@@ -58,17 +59,19 @@ func (d Double) Add(other ref.Val) ref.Val {
 
 // Compare implements traits.Comparer.Compare.
 func (d Double) Compare(other ref.Val) ref.Val {
-	otherDouble, ok := other.(Double)
-	if !ok {
+	switch ov := other.(type) {
+	case Double:
+		if math.IsNaN(float64(d)) || math.IsNaN(float64(ov)) {
+			return NewErr("NaN values cannot be ordered")
+		}
+		return compareDouble(d, ov)
+	case Int:
+		return compareDoubleInt(d, ov)
```

Comment:
Shouldn't we check for d being NaN here as well?

Reply:
Good point. Done.